### ISSUE TITLE
train_percent_check as a method for reducing train data size.

### ISSUE THREAD

**Issue by Nilanshrajput** [PyTorchLightning/pytorch-lightning#2179]
#### What is your question?
When we use `train_percent_check=x` while training does it uses the same x per cent of data in every epoch?
Can it be used as a method for making total data size short during training? 

#### Code

`Trainer(gpus=4,max_epochs=20,train_percent_check=.2,log_gpu_memory=True,weights_summary=None)`

**Comment by awaelchli** [comment]
> When we use train_percent_check=x while training does it uses the same x per cent of data in every epoch?

Yes, if you set shuffle=False in your dataloader, otherwise it will sample a different subset of data every epoch.

> Can it be used as a method for making total data size short during training?

Yes, that's what it is meant for. You can for example use it to sanity check that your model can overfit to the data.

**Comment by Nilanshrajput** [comment]
What I meant was, even id don't put shuffle=False, defining a train_percent should shuffle into that first x per cent of training data every epoch. So essentially it's like our training data is cut off to x per cent, and everything else remain the same.

**Comment by awaelchli** [comment]
Yes exactly, `train_percent_check=0.5` means use 50% of the dataset in the train_dataloader.
